Rendera/src/main.cpp

```diff
@@ -62,42 +71,84 @@ int main(){
     Camera cam(cam_origin, port_width, port_height, img_width, img_height, focal_length);
 
     //ppm file
-    std::ofstream image("sphere.ppm");
-    image << "P3\n";
-    image << img_width << " " << img_height << "\n";
-    image << max_col << "\n";
+    // std::ofstream image("sphere.ppm");
+    // image << "P3\n";
+    // image << img_width << " " << img_height << "\n";
+    // image << max_col << "\n";
+    
 
+    // for(int i=0; i<img_height; i++){
+    //     for(int j=0; j<img_width; j++){
 
-    for(int i=0; i<img_height; i++){
-        for(int j=0; j<img_width; j++){
+    //         //process each ray
+    //         color shade;
+    //         for(int sample=0; sample<50; sample++){
 
-            //process each ray
-            color shade;
-            for(int sample=0; sample<50; sample++){
+    //             ray ray_1 = cam.get_ray(j,i);
+
+    //             Intersection i;
+    //             bool hits = i.hit(s,ray_1);
+    //             color temp_shade = {1,1,1};
+                
+                
+    //             if(hits){
+
+    //                 vec3 hit_point = ray_1.fetch(i.dist_1()); 
+    //                 temp_shade = lighting(s.mat(), ray_1, s.normal(hit_point),hit_point,light_src, light_col);
+    //             }
+
+    //             if(sample==0){shade = temp_shade;}
+    //             shade = (shade*(sample) + temp_shade)* (1.0/(sample+1));
+
+    //         }
+    //         shade = shade.get_int();
+
+    //         //print result
+    //         image << shade.get_R() << " " << shade.get_G() << " " << shade.get_B()<<" " ;
+    //     }
+    // }
 
+    // image.close();
+    
+    // Jpeg and png
+    uint8_t* pixels = new uint8_t[img_width * img_height * 3];
```

Comment:
You should writing `std::uint8_t` because the `uint8_t` is in the `std` namespace. Also, a comment explaining why the array is of that dimension specifically would be helpful.